pkg/registry/seedmanagement/managedseed/storage/storage.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/pkg/apis/seedmanagement"
+	"github.com/gardener/gardener/pkg/registry/seedmanagement/managedseed"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+// REST implements a RESTStorage for ManagedSeed.
+type REST struct {
+	*genericregistry.Store
+}
+
+// ManagedSeedStorage implements the storage for ManagedSeeds and their status subresource.
+type ManagedSeedStorage struct {
+	ManagedSeed *REST
+	Status      *StatusREST
+}
+
+// NewStorage creates a new ManagedSeedStorage object.
+func NewStorage(optsGetter generic.RESTOptionsGetter) ManagedSeedStorage {
+	managedSeedRest, managedSeedStatusRest := NewREST(optsGetter)
+
+	return ManagedSeedStorage{
+		ManagedSeed: managedSeedRest,
+		Status:      managedSeedStatusRest,
+	}
+}
+
+// NewREST returns a RESTStorage object that will work with ManagedSeed objects.
+func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, *StatusREST) {
+	strategy := managedseed.NewStrategy()
+	statusStrategy := managedseed.NewStatusStrategy()
+
+	store := &genericregistry.Store{
+		NewFunc:                  func() runtime.Object { return &seedmanagement.ManagedSeed{} },
+		NewListFunc:              func() runtime.Object { return &seedmanagement.ManagedSeedList{} },
+		DefaultQualifiedResource: seedmanagement.Resource("managedseeds"),
+		EnableGarbageCollection:  true,
```

Comment:
> You will need it later on for fieldselectors for this field.

I don't quite get this, could you please explain?

Reply:
In the botanist, you're currently retrieving the managed seed, which references a given Shoot, via the `managedSeedName` field.
However, we agreed to drop this "back-reference" and therefore you will need an efficient way to lookup ManagedSeeds referencing a given seed. The best option for this will be to use a field selector, for which you will need an index here.

Reply:
Couldn't we simply avoid searching for the `ManagedSeed` in the botanist? With `APIServer` gone and the other changes discussed, there should hardly be any need for that, or am I missing something?

Reply:
Even if we can get rid of those things in the botanist, you will still need it in the admission plugin discussed above :)

Reply:
OK, got it.